src/palace/manager/api/controller/playtime_entries.py

```diff
@@ -16,9 +19,18 @@
 from palace.manager.sqlalchemy.model.collection import Collection
 from palace.manager.sqlalchemy.model.identifier import Identifier
 from palace.manager.sqlalchemy.model.library import Library
+from palace.manager.sqlalchemy.model.licensing import LicensePool
+from palace.manager.sqlalchemy.model.patron import Loan, Patron
 from palace.manager.sqlalchemy.util import get_one
 
 
+def resolve_loan_identifier(patron: Patron, loan: Loan | None) -> str:
+    def sha1(msg):
+        return hashlib.sha1(msg.encode()).hexdigest()
+
+    return sha1(f"loan: {loan.id}") if loan else sha1(f"patron:{patron.id}")
```

Comment:
I don't know if this helps so feel free to disregard, but we don't need to associate loans with patrons to satisfy this use case. It is sufficient to count the number of loans of the title that occurred during the reporting period, including instances where the same patron borrowed a title more than once.

Reply:
@tdilauro :  the use of the patron_id is just a fallback in the very rare case that the loan is not present.  For example if the playtime update did not happen for a long long time and the loan data had since been purged.  Otherwise the loan ID will be used in the vast majority of cases.   

We could, as you suggested discard individual entries.  I"m okay with making that change.

Reply:
I think it's better not to discard because we'll  lose some (if a small amount)  playtime and a few loans.  Or am I mistaken?